### DISCUSSION TITLE
Is receipt of mail ever silently blocked?

### DISCUSSION THREAD

**Discussion by roke-julian-lockhart** [anonaddy/anonaddy#670] [Q&A] [answered]
As I've explained in [a Reddit post — `reddit.com/user/rokejulianlockhart/comments/1ehld3w`](https://www.reddit.com/user/rokejulianlockhart/comments/1ehld3w/proton_mail_rejects_mail_using_spam_filters_that/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button):

<blockQuote>

As [a Reddit comment — `reddit.com/r/ProtonMail/comments/1e2kfpo/comment/ldng46g`](https://www.reddit.com/r/ProtonMail/comments/1e2kfpo/comment/ldng46g/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button) — explains, I recently had a message be rejected. If of interest, it was from Z-Lib, which is a service that provides literature. This isn't fundamentally problematic, except that the reason was that it was spam. If it was spam, it should have entered my spam folder, not been rejected.

I'm aware of why this feature exists — it means less strain on the servers. However, I can't trust a black-box algorithm to determine the probability that something is spam if the consequence of it being confident is outright rejection. Luckily, I was able to check [`app.addy.io/failed-deliveries`](https://app.addy.io/failed-deliveries) to recover it. However, anyone who doesn't utilize a middleman service with that capability shan't even know of its rejection, much less be able to recover it.

Perhaps this is common knowledge, but I didn't know of its existence until it occurred to me. Luckily, [a ProtonMail UserVoice feature request — `protonmail.uservoice.com/forums/284483-proton-mail/suggestions/17209559`](https://protonmail.uservoice.com/forums/284483-proton-mail/suggestions/17209559-option-to-disable-spam-filter#:~:text=Additionally%20a%20disable%20spam%20filter%20option%20would%20be%20excellent.%20A%20work%20around%20for%20adding%20a%20dedicated%20button%20might%20be%20a%20double%20wildcard%20whitelist:%20*@.) — for the ability to disable it exists. I suggest that those interested vote for it, and mark it as important or critical.

</blockQuote>

To confirm, does AnonAddy do this in any capacity? I do understand if it's a necessary evil, as long as there is very little room for doubt. I've asked because I need to prove it for [a Gaijin ZenDesk ticket — `support.gaijin.net/hc/requests/2601495`](https://support.gaijin.net/hc/requests/2601495).

**Comment by willbrowningme** [comment, marked as answer]
The addy.io mail servers use some spam protection measures as outlined here - [https://addy.io/faq/#how-do-you-prevent-spammers](https://addy.io/faq/#how-do-you-prevent-spammers).

This is required in order to protect the sending reputation of the IPs and domains that addy.io uses to deliver messages.

If all blatant spam messages were accepted and forwarded, addy.io would be seen as the source of this spam and be more likely to end up on blacklists or be blocked.

If a legitimate sender is being incorrectly blocked then you can contact me and I can whitelist them on the servers.

**Reply by roke-julian-lockhart** [reply]
> https://github.com/anonaddy/anonaddy/discussions/670#discussioncomment-10276018

@willbrowningme, thanks. I expect that rejected messages wouldn't be put into https://app.addy.io/failed-deliveries? If that's true, is there any way to know that a message was rejected? It would help me to be more confident if I knew that none were being rejected. Otherwise, mail delivery failure is a bit of a blind cat-and-mouse game.

Irrespective, I see some technologies listed there, like DMARC and SPF, of which it's stated that those which fail are rejected. However, doesn't AA instead merely forward the message with a warning banner, like https://github.com/anonaddy/anonaddy/discussions/668#discussion-7014874 depicts?

> ![image](https://github.com/user-attachments/assets/9ce8619e-a33f-4c39-a21f-a8a1058b2052)

**Reply by willbrowningme** [reply]
As in noted on the failed deliveries page:

> Only failed delivery attempts from the addy.io servers to your recipients (or reply/send attempts from your aliases) will be shown here. It will not show messages that failed to reach the addy.io server from some other sender.

I may be able to parse the mail logs in order to also show inbound rejections in the future.

Yes that banner is added when a message looks like spam or has a DMARC policy of `none`/`quarantine` that has failed. But if the domain had a DMARC policy of `reject` and it fails then it will be rejected.

**Reply by roke-julian-lockhart** [reply]
> https://github.com/anonaddy/anonaddy/discussions/670#discussioncomment-10303366

Thank you, @willbrowningme. That's a very reasonable policy. I've requested the ability to see inbound rejections at https://github.com/anonaddy/anonaddy/issues/672#issue-2459626001, and sent an e-mail in the meantime.
